pkg/chunk/inmemory_storage_client.go

```diff
@@ -162,15 +164,9 @@ func (m *MockStorage) BatchWrite(ctx context.Context, batch WriteBatch) error {
 			items = append(items, mockItem{})
 			copy(items[i+1:], items[i:])
 		} else {
-			// Return error if duplicate write and not metric name entry or series entry
-			itemComponents := decodeRangeKey(items[i].rangeValue)
-			keyType := itemComponents[3][0]
-			if keyType != metricNameRangeKeyV1 &&
-				keyType != seriesRangeKeyV1 &&
-				keyType != labelNamesRangeKeyV1 &&
-				keyType != labelSeriesRangeKeyV1 {
-				return fmt.Errorf("Dupe write")
-			}
+			// if duplicate write then just update the value
+			items[i].value = req.value
+			continue
```

Comment:
Could you explain why we need this change, please?

Reply:
The Add method in all the other stores does the updation of value if the record already exists. I wanted the behaviour of `MockStorage` to be the same.

Reply:
Ok then.